File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/BaloNoticeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.Affaire;
import com.jss.osiris.modules.osiris.quotation.model.BaloNotice;

public interface BaloNoticeService {

    public void updateBaloNotices();

    public List<BaloNotice> getBaloNoticeByAffaire(Affaire affaire);

}

File: client/src/app/modules/quotation/services/balo-notice.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { BaloNotice } from '../model/BaloNotice';

@Injectable({
  providedIn: 'root'
})
export class BaloNoticeService extends AppRestService<BaloNotice> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getBaloNoticeForAffaire(affaireId: number) {
    return this.getList(new HttpParams().set("affaireId", affaireId), "balo-notice/affaire");
  }
}
